src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEvent.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class DiagnosticEvent
+    {
+        private int _hitCount = 0;
+
+        public DateTime LastOccuredTimeStamp { get; set; }
+
+        public string Message { get; set; }
+
+        public string ErrorCode { get; set; }
+
+        public string HelpLink { get; set; }
+
+        public LogLevel Level { get; set; }
+
+        public int HitCount => _hitCount;
+
+        public void IncrementHitCount()
```

Comment:
This should probably live on the Repo itself. For some events (those in storage, for example), this won't be the way that we increment the hitcount.